src/common/snippets/src/generator.cpp

```diff
@@ -45,12 +45,11 @@ void Generator::generate(lowered::LinearIR& linear_ir, LoweringResult& result, c
     OV_ITT_TASK_NEXT(GENERATE, "::GetSnippet")
 
     // Note: some emitters use precompiled kernels. They need to be saved, so the kernels are accessible at runtime.
-    if (linear_ir.get_config().m_save_expressions) {
-        for (const auto& expr : linear_ir) {
-            const auto& emitter = expr->get_emitter();
-            if (uses_precompiled_kernel(emitter))
-                result.m_saved_emitters.emplace_back(emitter);
-        }
+    // Some debug capabilty also need emitters to be accessible at runtime.
+    for (const auto& expr : linear_ir) {
```

Comment:
@IvanNovoselov let me remind you our offline discussion basing on which I left [this recommendation](https://github.com/openvinotoolkit/openvino/pull/19935#discussion_r1390985117): we agreed that keeping the flag just to avoid one loop at the compilation stage is an overkill. Also, it was decided to handle the segfault detector case inside `uses_precompiled_kernel` method